### PR TITLE
recalculate_order was not being called in 1817 some times

### PR DESCRIPTION
**Pull request by daniel-sousa-me**
It now also removes corporations in liquidation

### CODE REVIEW

**Comment by daniel-sousa-me** [comment]
This is high priority, because it's breaking games

File: lib/engine/step/train.rb [hunk 1]
```diff
@@ -90,7 +90,6 @@ def process_sell_shares(action)
         @last_share_sold_price = action.bundle.price_per_share
         super
         @corporations_sold << action.bundle.corporation
-        @round.recalculate_order
```
Comment (jenf) [inline lib/engine/step/train.rb:93]: This needs to be left

File: lib/engine/round/operating.rb [hunk 2]
```diff
@@ -74,6 +74,8 @@ def recalculate_order
         # to change order. Re-sort only them.
         index = @entity_index + 1
         @entities[index..-1] = @entities[index..-1].sort if index < @entities.size - 1
+
+        @entities.pop while @entities.last.corporation? && @entities.last.share_price.liquidation?
```
Comment (jenf) [inline lib/engine/round/operating.rb:78]: would this break if entities is empty?

File: lib/engine/step/train.rb (hunk 1 above)
Comment (daniel-sousa-me) [reply lib/engine/step/train.rb:93]: Doesn't `super` call the one in `emergency_money.rb`?

File: lib/engine/round/operating.rb (hunk 2 above)
Comment (daniel-sousa-me) [reply lib/engine/round/operating.rb]: It does... I don't know how I missed that :disappointed:

**Review by jenf — APPROVED** [review]

**Pull request merged**